### ISSUE TITLE
istioctl kube-inject broken in 1.0.x when applied against 1.1 Istio install

### ISSUE THREAD

**Issue by esnible** [istio/istio#11565]
To reproduce:

```
curl -L https://git.io/getLatestIstio | sh -
./istio-1.0.5/bin/istioctl kube-inject -f istio-1.0.5/samples/httpbin/httpbin.yaml 
Error: failed to convert to proto. unknown field "tracing" in v1alpha1.ProxyConfig
```

Same behavior verified on Linux and OSX for 1.0.5, and for OSX for 1.0.0, .1, .2, .3, and .4.  Did not test Linux other than 1.0.5.

Recording this as a defect because the error message surprised me and I didn't see it documented elsewhere.  Do we expect back-level _istioctl_ to be able to give a good reason for failing when used on a 1.1 cluster?

**Comment by ozevren** [comment]
@ayj, any thoughts on doing version checks in istioctl to improve the experience?

**Comment by ayj** [comment]
I don't think MeshConfig proto includes the version. We would need to guess based on the installed CP version and differentiate new fields (that were added in newer versions) from malformed/mispelled config.   

We addressed this in Pilot by allowing unknown fields (see https://github.com/istio/istio/pull/9792). This doesn't differentiate misspelled fields vs. the backwards compatibility case. I tried to fix this once in kube-inject (see https://github.com/istio/istio/pull/1751), but there was disagreement about whether it was the correct solution.
